gateway/fields.go

```diff
@@ -4,37 +4,16 @@ import (
 	"context"
 	"net/http"
 
-	"github.com/grpc-ecosystem/grpc-gateway/runtime"
 	"github.com/infobloxopen/atlas-app-toolkit/query"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/metadata"
 )
 
-//SetFieldSelection sets op.FieldSelection to gRPC metadata
-func SetFieldSelection(ctx context.Context, fields *query.FieldSelection) error {
-	fieldsStr := fields.GoString()
-	md := metadata.Pairs(
-		runtime.MetadataPrefix+fieldsMetaKey, fieldsStr,
-	)
-	return grpc.SetHeader(ctx, md)
-}
-
-//FieldSelection extracts op.FieldSelection from gRPC metadata
-func FieldSelection(ctx context.Context) *query.FieldSelection {
-	fields, ok := Header(ctx, fieldsMetaKey)
-	if !ok {
-		return nil
-	}
-	return query.ParseFieldSelection(fields)
-}
-
 //retainFields function extracts the configuration for fields that
 //need to be ratained either from gRPC response or from original testRequest
 //(in case when gRPC side didn't set any preferences) and retains only
 //this fields on outgoing response (dynmap).
 func retainFields(ctx context.Context, req *http.Request, dynmap map[string]interface{}) {
-	fieldsStr, ok := Header(ctx, fieldsMetaKey)
-	if !ok && req != nil {
+	fieldsStr := ""
+	if req != nil {
 		//no fields in gprc response -> try to get from original testRequest
```

Comment:
That retainFields is called by ForwardMessage where request is available as http.Request, so don't have access to message collection fields. As result, _fields will be parsed only from http.request